libp2p/src/main/kotlin/io/libp2p/mux/yamux/YamuxHandler.kt

```diff
@@ -87,11 +91,23 @@ open class YamuxHandler(
                     writeAndFlushFrame(YamuxFrame(msg.id, YamuxType.WINDOW_UPDATE, YamuxFlag.ACK.asSet, 0))
                 }
 
+                YamuxFlag.ACK in msg.flags -> {
+                    if (outbound) {
+                        acknowledged.set(true)
+                    }
+                }
+
                 YamuxFlag.FIN in msg.flags -> onRemoteDisconnect(msg.id)
                 YamuxFlag.RST in msg.flags -> onRemoteClose(msg.id)
             }
         }
 
+        private fun acknowledgeInboundStreamIfNeeded() {
+            if (!outbound) {
+                acknowledged.compareAndSet(false, true)
```

Comment:
nit: just `set()` seems to yield the same result?
BTW: I believe we basically don't need any thread safe code here. All the methods should be invoked on the same Netty event thread. Anyways the overall `YamuxHandler` code is not thread safe globally.